src/components/HomeScreen.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import {
+  TextInput,
+  TouchableOpacity,
+  Button,
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { connect } from 'react-redux';
+
+const { navigation } = this.props.navigation;
+
+class HomeScreen extends Component {
+  state = {
+    email: "",
+    password: "",
+  }
+  
+
+  onLogin (){
+    const user = {
+      email: "",
+      password: ""
+    };
+    user[email] = this.state.email;
+    user[password] = this.state.password;
+    this.props.onLogin(user);
+  } 
+
+
+  render(){
+    return (
```

Comment:
O **this.props** só está disponivel dentro do contexto do componente porque vai usar as propriedades que sao mapeadas para o props, portanto tens que o definir dentro da funcao do render
```suggestion
  render(){
    const { navigation } = this.props.navigation;
    return (
```